go.mod

```diff
@@ -1,40 +1,103 @@
 module github.com/giantswarm/cluster-apps-operator
 
-go 1.16
+go 1.17
 
 require (
-	github.com/giantswarm/apiextensions/v3 v3.38.0
-	github.com/giantswarm/apptest v0.12.0
+	github.com/giantswarm/apiextensions-application v0.0.0-20211118184941-0e4a8fce3437
+	github.com/giantswarm/apptest v0.12.1-0.20211119165242-9255e2e76c06
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/exporterkit v0.2.1
-	github.com/giantswarm/k8sclient/v5 v5.12.0
+	github.com/giantswarm/k8sclient/v6 v6.0.0
 	github.com/giantswarm/k8smetadata v0.6.0
 	github.com/giantswarm/microendpoint v0.2.0
 	github.com/giantswarm/microerror v0.3.0
 	github.com/giantswarm/microkit v0.2.2
 	github.com/giantswarm/micrologger v0.5.0
-	github.com/giantswarm/operatorkit/v5 v5.0.0
-	github.com/giantswarm/resource/v3 v3.0.2
+	github.com/giantswarm/operatorkit/v6 v6.0.0
+	github.com/giantswarm/release-operator/v2 v2.4.1-0.20211119182623-079d1f7a4e8c
+	github.com/giantswarm/resource/v3 v3.0.3-0.20211119182905-30ba680fdb1b
 	github.com/patrickmn/go-cache v2.1.0+incompatible
 	github.com/prometheus/client_golang v1.11.0
 	github.com/spf13/viper v1.9.0
-	k8s.io/api v0.20.12
-	k8s.io/apimachinery v0.20.12
-	k8s.io/client-go v0.20.12
-	sigs.k8s.io/cluster-api v0.4.4
-	sigs.k8s.io/cluster-api-provider-azure v0.5.4
-	sigs.k8s.io/controller-runtime v0.6.5
+	k8s.io/api v0.21.4
+	k8s.io/apimachinery v0.21.4
+	k8s.io/client-go v0.21.4
+	sigs.k8s.io/cluster-api v0.3.13
+	sigs.k8s.io/controller-runtime v0.9.7
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
```

Comment:
A bit off topic. I know this is from Go 1.17 but will we be able to remove these indirect references?

Makes go.mod harder to follow compared to 1.16

Reply:
As far as I know it's required in 1.17. We could keep the go version in `go.mod` at 1.16, but it would just be a matter of time...